bugsnag-android-core/src/main/java/com/bugsnag/android/internal/InternalMetricsImpl.kt

```diff
@@ -7,16 +7,21 @@ class InternalMetricsImpl : InternalMetrics {
     private val callbackCounts = hashMapOf<String, Int>()
 
     override fun toJsonableMap(): Map<String, Any> {
-        return mapOf(
-            "config" to configDifferences,
-            "callbacks" to allCallbacks()
-        )
+        val callbacks = allCallbacks()
+
+        return listOfNotNull(
+            if (configDifferences.isNotEmpty()) "config" to configDifferences else null,
+            if (callbacks.isNotEmpty()) "callbacks" to callbacks else null,
+        ).toMap()
```

Comment:
I feel like if we're going to keep using this pattern, we should probably create our own helpers for it. Not for this PR but something to consider.